test/e2e/aadpodidentity_test.go

```diff
@@ -446,6 +446,65 @@ var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
 	// 	removeUserAssignedIdentityFromCluster(nodeList, clusterIdentity)
 	// })
 
+	It("should cleanup iptable rules after deleting aad-pod-identity", func() {
+		// create the helper ssh daemon for validating the iptable rules
+		cmd := exec.Command("kubectl", "apply", "-f", "template/busyboxds.yaml")
+		_, err := cmd.CombinedOutput()
+		Expect(err).NotTo(HaveOccurred())
+
+		time.Sleep(90 * time.Second)
```

Comment:
Added 90s wait time because it's possible nmi pods are just being created. It takes 60s after the pod startup before the iptables entries are added since the ticker is configured with 60s update duration.  We might need to change this to create the ticker with instant first tick to prevent any race conditions.

Reply:
Could we add a check to ensure that the rules have been indeed added instead of the sleep to make it more deterministic. Perhaps in a future PR, but could make our tests more deterministic ?

Reply:
That makes sense. I'll open an issue to track that change.